public/app/features/dashboard/components/DashboardSettings/AutoRefreshIntervals.test.tsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import { defaultIntervals } from '@grafana/ui/src/components/RefreshPicker/RefreshPicker';
+
+import { AutoRefreshIntervals, getValidIntervals, Props, validateIntervals } from './AutoRefreshIntervals';
+import { TimeSrv } from '../../services/TimeSrv';
+
+const setupTestContext = (options: Partial<Props>) => {
+  const defaults: Props = {
+    renderCount: 0,
+    refreshIntervals: ['1s', '5s', '10s'],
+    onRefreshIntervalChange: jest.fn(),
+    getIntervalsFunc: intervals => intervals,
+    validateIntervalsFunc: () => null,
+  };
+
+  const props = { ...defaults, ...options };
+  const { rerender } = render(<AutoRefreshIntervals {...props} />);
+
+  return { rerender, props };
+};
+
+describe('AutoRefreshIntervals', () => {
+  describe('when component is mounted with refreshIntervals', () => {
+    it('then supplied intervals should be shown', () => {
+      setupTestContext({ getIntervalsFunc: () => ['5s', '10s'] }); // remove 1s entry to validate we're calling getIntervalsFunc
+
+      const input = (screen.getByRole('textbox') as unknown) as HTMLInputElement;
```

Comment:
Wondering why `unknown` is necessary here? I remember writing similar tests and just casting as `HTMLInputElement` was enough. 

Also if the goal is to test input value after component has been rendered, i.e. there was no manual `onChange` trigger via `fireEvent`, `toHaveAttribute` can be used, which doesn't require type casting: ` expect(screen.getByRole('textbox')).toHaveAttribute('value', '5s,10s');`

Reply:
Great feedback @Clarity-89 , I actually read up a lot on RTL last evening so I already had a long list of things to change. This is one of those. I'll be right back, again thank you!